ctapipe/visualization/mpl_array.py

```diff
@@ -232,28 +234,33 @@ def set_vector_hillas(self, hillas_dict, length, time_gradient, angle_offset):
         rot_angle_ellipse = np.zeros(self.subarray.num_tels) * u.deg
 
         for tel_id, params in hillas_dict.items():
+
             idx = self.subarray.tel_indices[tel_id]
             rho[idx] = u.Quantity(length, u.m)
 
+            psi = event.dl1.tel[tel_id].parameters.core.psi
+
             if time_gradient[tel_id] > 0.01:
-                params.psi = Angle(params.psi)
                 angle_offset = Angle(angle_offset)
-                rot_angle_ellipse[idx] = params.psi + angle_offset + 180 * u.deg
+                rot_angle_ellipse[idx] = psi + angle_offset + 180 * u.deg
             elif time_gradient[tel_id] < -0.01:
-                rot_angle_ellipse[idx] = params.psi + angle_offset
+                rot_angle_ellipse[idx] = psi + angle_offset
             else:
                 rho[idx] = 0 * u.m
 
         self.set_vector_rho_phi(rho=rho, phi=rot_angle_ellipse)
 
-    def set_line_hillas(self, hillas_dict, range, **kwargs):
+    def set_line_hillas(self, event, hillas_dict, range, **kwargs):
         """
```

Comment:
Same here: do not pass the whole event into the function, just the bits of it you need , as with hillas_dict.